c2rust-transpile/src/c_ast/mod.rs

```diff
@@ -888,76 +885,90 @@ impl TypedAstContext {
-                            } else {
-                                None
-                            }
-                        }
-                        CExprKind::Binary(_ty, op, lhs, rhs, _, _) => {
-                            let rhs_type_id =
-                                self.ast_context.c_exprs[&rhs].kind.get_qual_type().unwrap();
-                            let lhs_kind = &self.ast_context.c_exprs[&lhs].kind;
-                            let lhs_type_id = lhs_kind.get_qual_type().unwrap();
-
-                            let lhs_resolved_ty = self.ast_context.resolve_type(lhs_type_id.ctype);
-                            let rhs_resolved_ty = self.ast_context.resolve_type(rhs_type_id.ctype);
-
-                            let neither_ptr = !lhs_resolved_ty.kind.is_pointer()
-                                && !rhs_resolved_ty.kind.is_pointer();
-
-                            if op.all_types_same() && neither_ptr {
-                                if CTypeKind::PULLBACK_KINDS.contains(&lhs_resolved_ty.kind) {
-                                    Some(lhs_type_id)
-                                } else {
-                                    Some(rhs_type_id)
-                                }
-                            } else if op == BinOp::ShiftLeft || op == BinOp::ShiftRight {
-                                Some(lhs_type_id)
                             } else {
-                                return;
+                                Some(rhs_type_id)
                             }
+                        } else if op == BinOp::ShiftLeft || op == BinOp::ShiftRight {
+                            Some(lhs_type_id)
+                        } else {
+                            return;
                         }
-                        CExprKind::Unary(_ty, op, e, _idk) => op.expected_result_type(
-                            self.ast_context,
-                            self.ast_context.c_exprs[&e].kind.get_qual_type().unwrap(),
-                        ),
-                        CExprKind::Paren(_ty, e) => {
-                            self.ast_context.c_exprs[&e].kind.get_qual_type()
-                        }
-                        _ => return,
-                    };
-                    if let (Some(ty), Some(new_ty)) = (
-                        self.ast_context
-                            .c_exprs
-                            .get_mut(&e)
-                            .and_then(|e| e.kind.get_qual_type_mut()),
-                        new_ty,
-                    ) {
-                        *ty = new_ty;
-                    };
-                }
+                    }
+                    CExprKind::Unary(_ty, op, e, _idk) => op.expected_result_type(
+                        self.ast_context,
+                        self.ast_context.c_exprs[&e].kind.get_qual_type().unwrap(),
+                    ),
+                    CExprKind::Paren(_ty, e) => self.ast_context.c_exprs[&e].kind.get_qual_type(),
```

Comment:
Why did this need to change? The new line is too long, the old block made it more readable.

Reply:
I just decreased the indentation one level, which generally makes things more readable, but this particular line is up to rustfmt.